File: repos_estudiantes_TARQUI_primera_entrega/LUNA_CRUZ_CAMILA_MILENKA/Ejercicio7/Celular.java
package Ejercicio7;
//Crea un clase Celular con espacios para 20 aplicaciones
//o 1024MB de espacio
public class Celular {
    private String[] aplicaciones; 
    private int espacioDisponible, bateria, cantidadAplicaciones; 

    public Celular() 
    {
        this.aplicaciones = new String[20];  
        this.espacioDisponible = 1024;  
        this.bateria = 100;  
        this.cantidadAplicaciones = 0;  
    }

    public boolean InstalarNuevaAplicacion(String nombre, int tamaño) 
    {
        if (cantidadAplicaciones < 20 && espacioDisponible >= tamaño) 
        {
            aplicaciones[cantidadAplicaciones] = nombre;
            cantidadAplicaciones++;
            espacioDisponible -= tamaño;  
            System.out.println("Aplicacion " + nombre + " se instalo con exito.");
            return true;
        } else {
            System.out.println("Aplicacion no instalada. Falta espacio");
            return false;
        }
    }

    public void utilizarAplicacion(int minutosUso, int tamañoAplicacion) 
    {
        if (bateria <= 0) 
        {
            System.out.println("Apagado");
            return;
        }
        double porcentajeBateria = 0;
        if (tamañoAplicacion > 250) 
        {
            porcentajeBateria = 5;  
        } 
        else
            if (tamañoAplicacion > 100) 
            {
            porcentajeBateria = 2;  
            } 
            else {
            porcentajeBateria = 1;  
            }
        int bloquesDe10Minutos = minutosUso / 10;
        double consumoTotal = porcentajeBateria * bloquesDe10Minutos;

        bateria -= consumoTotal;
        if (bateria < 0) 
        {
            bateria = 0; 
        }
        System.out.println("Se uso la aplicacion durante " + minutosUso + " minutos.");
    }

    public void mostrarPorcentajedeBateriaRestante() {
        System.out.println("Bateria restante: " + bateria + "%");
    }
    
    public static void main(String[] args) {
        Celular celular = new Celular();
        
        System.out.println("------APLICACIONES---------");
        celular.InstalarNuevaAplicacion("Tik Tok", 900); 
        celular.InstalarNuevaAplicacion("Instagram", 622);
        celular.InstalarNuevaAplicacion("Canva", 50);

        System.out.println("---------------------------------");
        
        System.out.println("-----USO Y BATERIA-----");
        celular.utilizarAplicacion(30, 100); 
        celular.mostrarPorcentajedeBateriaRestante();  

        celular.utilizarAplicacion(60, 200); 
        celular.mostrarPorcentajedeBateriaRestante();  

        celular.utilizarAplicacion(100, 300); 
        celular.mostrarPorcentajedeBateriaRestante();  
    }
    
}

File: repos_estudiantes_TARQUI_primera_entrega/LUNA_CRUZ_CAMILA_MILENKA/Ejercicio7/Celular.py
#Crea un clase Celular con espacios para 20 aplicaciones
#o 1024MB de espacio

class Celular:
    def __init__(self):
        self.aplicaciones = [None] * 20  # lista de aplicaciones
        self.espacio_disponible = 1024  # en MB
        self.bateria = 100  # en porcentaje
        self.cantidad_aplicaciones = 0

    def instalar_nueva_aplicacion(self, nombre, tamano):
        if self.cantidad_aplicaciones < 20 and self.espacio_disponible >= tamano:
            self.aplicaciones[self.cantidad_aplicaciones] = nombre
            self.cantidad_aplicaciones += 1
            self.espacio_disponible -= tamano
            print(f"Aplicación {nombre} se instaló con éxito.")
            return True
        else:
            print("Aplicación no instalada. Falta espacio.")
            return False

    def utilizar_aplicacion(self, minutos_uso, tamano_aplicacion):
        if self.bateria <= 0:
            print("Celular apagado.")
            return

        if tamano_aplicacion > 250:
            porcentaje_bateria = 5
        elif tamano_aplicacion > 100:
            porcentaje_bateria = 2
        else:
            porcentaje_bateria = 1

        bloques_de_10_minutos = minutos_uso // 10
        consumo_total = porcentaje_bateria * bloques_de_10_minutos

        self.bateria -= consumo_total
        if self.bateria < 0:
            self.bateria = 0

        print(f"Se usó la aplicación durante {minutos_uso} minutos.")
    
    def mostrar_porcentaje_bateria_restante(self):
        print(f"Batería restante: {self.bateria}%")

if __name__ == "__main__":
    celular = Celular()

    print("------APLICACIONES---------")
    celular.instalar_nueva_aplicacion("Tik Tok", 900)
    celular.instalar_nueva_aplicacion("Instagram", 622)
    celular.instalar_nueva_aplicacion("Canva", 50)

    print("---------------------------------")
    
    print("-----USO Y BATERÍA-----")
    celular.utilizar_aplicacion(30, 100)
    celular.mostrar_porcentaje_bateria_restante()

    celular.utilizar_aplicacion(60, 200)
    celular.mostrar_porcentaje_bateria_restante()

    celular.utilizar_aplicacion(100, 300)
    celular.mostrar_porcentaje_bateria_restante()